src/coins.cpp

```diff
@@ -206,10 +204,9 @@ bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn
                 // and mark it as dirty.
                 itUs = cacheCoins.try_emplace(it->first).first;
                 CCoinsCacheEntry& entry{itUs->second};
-                if (erase) {
-                    // The `move` call here is purely an optimization; we rely on the
-                    // `mapCoins.erase` call in the `for` expression to actually remove
-                    // the entry from the child map.
+                if (cursor.WillErase(*it)) {
+                    // Since this entry will be erased,
+                    // we can move the coin into us instead of copying it
```

Comment:
nit: "move coin into us" sounds a bit weird.
And now that the code is cleaner, does the comment add anything that the code doesn't already say?

* `Since this entry will be erased` -> `if (cursor.WillErase(*it)) {`
* `we can move the coin into us instead of copying it` -> `std::move(it->second.coin)` vs `it->second.coin`

Reply:
You might be right here, but this function in particular is very heavily commented due to its importance. I think modifying the comment is better than just removing it.